### ISSUE TITLE
Implement methods for multiplex graphs

### ISSUE THREAD

**Issue by TomKellyGenetics** [TomKellyGenetics/leiden#7]
It is possible to call multiplex graphs in python as detailed here: https://leidenalg.readthedocs.io/en/latest/multiplex.html

Once the bug in leidenalg 0.8.2 is addressed (https://github.com/vtraag/leidenalg/issues/50) it can be supported in R.

Here is a working demo of calling it in R via reticulate on a list of python graph objects (generated by leiden.igraph).

```
leidenalg <- import("leidenalg", delay_load = TRUE)
leidenalg <- import("leidenalg", delay_load = TRUE)
self.optimiser = leidenalg$Optimiser()
graph_list <- r_to_py(list(graph1, graph2, graph3))
part <- leidenalg$find_partition_multiplex(graph_list, partition_type = leidenalg$CPMVertexPartition)
names(part) <- c("membership", "improv")
partition <- part$membership+1
partition
```

It will be possible to define a dedicated internal function (similar to "find_partitions") specifically for multiplex graphs based on this example. The question is whether to add and S3 method for "leiden.list" or make it a separate multiplex function.

R has multiple supported classes for inputs for multiplex graphs:

1. a list of igraph objects as demonstrated

1. a "Semigroup" in the "multiplex" package (basically a list of arrays)

1. the "network" package which supports the "network" class (which has an "is.multiplex" function)

1. the [mully](https://github.com/frankkramer-lab/mully) package on CRAN has a "mully" class which depends on igraph ans supports an "is.mully" function.

1. the "multinet" package has the "RMLNetwork" class (can be converted to igraph with as.igraph.Rcpp_RMLNetwork)

1. the [MUNA](https://github.com/Issamfalih/MUNA/) package is only on GitHub but extends igraph objects for multiplex graphs. The ""Multiplex" class has a "layers" slot which is a list of igraph objects.

1. the [mplex](https://github.com/Achab94/mplex) package (not maintained) has a "multiplex" class.

These can be converted by passing to the adjacency matrix, although an igraph method would be preferable..

**Comment by vtraag** [comment]
@TomKellyGenetics, the new version, including the bugfix you contributed (https://github.com/vtraag/leidenalg/pull/51) is now released as version 0.8.3 and is available from PyPI.

**Comment by TomKellyGenetics** [comment]
Thanks that's great to know. I'll add examples to the vignettes before it goes on CRAN.

If anyone wants to use it before then of course, all they need to do is install leidenalg 0.8.3 and use devtools for R.

Note: this also depends on the development version of r-igraph as it calls `igraph::cluster_leiden` which is not available in igraph 1.2.6 on CRAN. See https://github.com/TomKellyGenetics/leiden/issues/1

```
devtools::install_github("TomKellyGenetics/leiden", ref = "dev")
```

**Comment by TomKellyGenetics** [comment]
A branch which does _not_ depend on the development version of r-igraph is available here.

```
devtools::install_github("TomKellyGenetics/leiden", ref = "multiplex")
```

**Comment by vtraag** [comment]
Note that you should be able to install the development version of R `igraph` using `remotes::install_github("igraph/rigraph@master")`, see also  https://github.com/igraph/rigraph/issues/423#issuecomment-731206758

**Comment by TomKellyGenetics** [comment]
Yeah it should work but I had to install it from source and troubleshoot compiling errors. That might have been me missing dependencies at the time due to recently updating Mac OS though. I have managed to test the dev version of igraph and it works great! 

It wasn't difficult to cherry-pick over the functions for multiplex graphs into the master branch that doesn't depend on cluster_leiden yet. Hopefully igraph isn't so difficult for others to install from github but bearing in mind R users may have trouble installing the C libraries. This also means this branch is CRAN compatible with a minor release.

**Comment by vtraag** [comment]
> I had to install it from source and troubleshoot compiling errors

If you could report those errors during installation/compilation in the `rigraph` repository, that would be great.

**Comment by TomKellyGenetics** [comment]
> If you could report those errors during installation/compilation in the `rigraph` repository, that would be great.
 
Yeah I will if there's a problem with package. In this case I resolved it by reinstalling gcc and gfortran. Turns out I couldn't even compile hello_world.c with the missing headers.

**Comment by TomKellyGenetics** [comment]
This is now implemented in version 0.3.6 in the "multiplex" branch and version 0.4.0 in the dev branch which supports `igraph::cluster_leiden`. The `max_comm_size` parameter is also supported by these releases. #8 

All this needs is a vignette for multiplex graphs, based on the example working in the tests this should not be difficult. While plotting multiplexed graphs for this I discovered a bug in `graphsim:plot_directed` (with unnamed vertices) in graphsim 1.0.1 and updating the vignettes in this package has delayed progress on this.
